src/main/java/net/dv8tion/jda/internal/entities/EntityBuilder.java

```diff
@@ -474,8 +473,10 @@ public void updateUser(UserImpl userObj, DataObject user)
     {
         String oldName = userObj.getName();
         String newName = user.getString("username");
+        String oldGlobalName = userObj.getName();
```

Comment:
I think this causes the UserUpdateGlobalNameEvent to be repeatedly fired when the username doesn't match the display name.

```suggestion
        String oldGlobalName = userObj.getGlobalName();
```